test/integration/ordinary/itest_slc.py

```diff
@@ -125,15 +139,29 @@ def test_deploy(sample_slc: ScriptLanguageContainer, itde):
 
 @pytest.mark.dependency(name="append_custom_packages", depends=["deploy_slc"])
 def test_append_custom_packages(
-    sample_slc: ScriptLanguageContainer, custom_packages: list[tuple[str, str, str]]
+    sample_slc: ScriptLanguageContainer,
+    custom_packages: list[tuple[str, str, str]],
+    package_manager: PackageManager,
 ):
-    sample_slc.append_custom_packages(
-        [PipPackageDefinition(pkg, version) for pkg, version, _ in custom_packages]
-    )
-    with open(sample_slc.custom_pip_file) as f:
-        pip_content = f.read()
-        for custom_package, version, _ in custom_packages:
-            assert f"{custom_package}|{version}" in pip_content
+    if package_manager == PackageManager.PIP:
```

Comment:
Split into two tests and make each one skip when the package manager is not the expected one

Reply:
Then using dependent tests , is maybe not the correct strategy; this sounds more like putting most of the stuff into fixtures and only put the asserts into different tests.